internal/ingress/controller/store/store.go

```diff
@@ -301,6 +310,51 @@ func (s k8sStore) GetServiceEndpoints(key string) (*corev1.Endpoints, error) {
 	return s.listers.Endpoint.ByKey(key)
 }
 
+func (s *k8sStore) annotationToNodeLabelObj(nodeLabelAnnotation string) *[]NodeLabel {
```

Comment:
1. it should return []NodeLabel instead of *[]NodeLabel to ease GC pressure.  (minimal comment)